sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/ArchiveBasedAnalysisInputLocation.java

```diff
@@ -99,8 +99,13 @@ public Optional<JavaSootClassSource> getClassSource(@Nonnull ClassType type, @No
     try {
       FileSystem fs = fileSystemCache.get(path);
       final Path archiveRoot = fs.getPath("/");
+      JavaClassType javaClassType = null;
+      if (type instanceof JavaClassType)
+        javaClassType = (JavaClassType) type;
+      else
+        javaClassType = new JavaClassType(type.getClassName(), type.getPackageName());
       return getClassSourceInternal(
-          (JavaClassType) type, archiveRoot, new AsmJavaClassProvider(view));
+          javaClassType, archiveRoot, new AsmJavaClassProvider(view));
```

Comment:
what happened here?

Reply:
We should address this issue in whole in another PR - please add a comment that this is a temporary hack